pkg/apiserver/internal/v1/service.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2020 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	v1 "github.com/kubermatic/kubecarrier/pkg/apiserver/api/v1"
+	"github.com/kubermatic/kubecarrier/pkg/apiserver/internal/util"
+)
+
+type serviceServer struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+var _ v1.ServicesServer = (*serviceServer)(nil)
+
+func NewServicesServer(c client.Client, scheme *runtime.Scheme) v1.ServicesServer {
+	return &serviceServer{
+		client: c,
+		scheme: scheme,
+	}
+}
+func (o serviceServer) Create(ctx context.Context, req *v1.ServiceCreateRequest) (res *v1.Service, err error) {
+	obj := &unstructured.Unstructured{}
+
+	gvk := o.gvkFromService(req.Service, req.Version)
+	obj.SetGroupVersionKind(gvk)
+	val := map[string]interface{}{}
+	if err := json.Unmarshal([]byte(req.Spec.Spec), &val); err != nil {
+		return nil, status.Errorf(codes.Internal, "creating services: spec should by type of map[string]intreface{}")
+	}
+	if err := unstructured.SetNestedMap(obj.Object, val, "spec"); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("creating services: %s", err.Error()))
+	}
+	if err := util.SetMetadata(obj, req.Spec.Metadata); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("creating services: %s", err.Error()))
+	}
+	if err := o.client.Create(ctx, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("creating services: %s", err.Error()))
+	}
+	if err = o.client.Get(ctx, types.NamespacedName{
```

Comment:
Why do you need to get the object after creating it?

Reply:
thanks, didn't know that `Create` update `obj`